dspace-api/src/main/java/org/dspace/content/Site.java

```diff
@@ -83,11 +90,7 @@ public static String getSiteHandle()
     public static DSpaceObject find(Context context, int id)
         throws SQLException
     {
-        if (theSite == null)
-        {
-            theSite = new Site();
-        }
-        return theSite;
+        return new Site(context);
```

Comment:
I noticed that other DSOs follow this pattern ([from Item.java](https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/content/Item.java#L130)):


        // First check the cache
        Item fromCache = (Item) context.fromCache(Item.class, id);

        if (fromCache != null)
        {
            return fromCache;
        }
        // otherwise construct a new one

Would it make sense to follow the same approach here? I guess it might not be so important since Site doesn't (yet) go through to the db in the way other DSOs do?